src/auth/Readme.md

```diff
@@ -0,0 +1,266 @@
+  storage: window.localStorage,
+});
+```
+
+## How do I use the auth without prop drilling?
+
+#### **`SaleorAuthProvider({ saleorAuthClient, isAuthenticating, children }: PropsWithChildren<UseSaleorAuthClient>) => JSX.Element`**
+
+The hook is supposed to be used only in the root of your project. If you'd like to access the client methods or isAuthenticating prop down the tree without extensive prop drilling, you can use `SaleorAuthProvider` passing it down the return value of useSaleorAuthClient hook.
+
+```jsx
+const saleorAuthClientData = useSaleorAuthClient(authClientProps)
+
+<SaleorAuthProvider {...saleorAuthClientData}>
+    {...}
+</SaleorAuthProvider>
+```
+
+---
+
+#### **`useSaleorAuthContext() => SaleorAuthContextConsumerProps`**
+
+The provider also equips you with `useSaleorAuthContext` hook
+
+```javascript
+const { isAuthenticating, signIn, signOut, checkoutSignOut } = useSaleorAuthContext();
+```
+
+## How do I tell my graphql client to refresh queries on signIn / signOut?
+
+#### **`useAuthChange({ onSignedIn, onSignedOut }: UseAuthChangeProps) => void`**
+
+Once the successful sign in or sign out happens, as well as failed authentication (for e.g when refresh token is expired and there's no way to obtain a new access token from the api) you'll probably want to trigger some actions. Hence we're provided a `useAuthChange` hook that'll listen to storage changes and trigger callbacks.
+
+```javascript
+  useAuthChange({
+    onSignedOut: () => { /* client invalidate some queries */},
+    onSignedIn: () => { /* client invalidate some queries  */}
+    },
+  });
+```
+
+## How do I sign in?
+
+#### **`SaleorAuthClient.signIn: ({ email, password }: TokenCreateVariables) => Promise<TokenCreateResponse>`**
+
+SaleorAuthClient returns a `signIn` method. You can also access it via the **useSaleorAuthContext** hook.
+
+```javascript
+const { signIn } = useSaleorAuthContext();
+
+const response = await signIn({
+  email: "example@mail.com",
+  passowrd: "password",
+});
+```
+
+## How do I sign out?
+
+#### **`SaleorAuthClient.signOut: () => void`**
+
+This method will remove access and refresh tokens from SaleorAuthClient state and storage. It'll also trigger sign out event, which in turn will trigger [onSignOut method of useAuthChange hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout)
+
+```javascript
+const { signOut } = useSaleorAuthContext();
+
+signOut();
+```
+
+⚠️ For checkout sign out see [signing out in checkout](#how-do-i-sign-out-in-checkout)
+
+## How do I sign out in checkout?
+
+#### **`SaleorAuthClient.checkoutSignOut: ({ checkoutId }: CustomerDetachVariables) => Promise<CustomerDetachResponse>`**
+
+On top of the regular sign out login, in checkout we need to start signing out process with detaching customer from checkout. Since detach requires a signed in user, it'll happen first and removing tokens from state / storage will only happen if the mutation returned success:
+
```

Comment:
```suggestion
When dealing with authentication in the checkout, we need to start the signing-out process by detaching the customer from checkout. Since it requires the user to be signed in, it must be executed first. If the mutation succeeds, the tokens from the state/storage will be removed.

```